src/core/services/administrator_service.py

```diff
@@ -1,12 +1,34 @@
+from uuid import UUID
+
 from fastapi import Depends
 
+from src.api.request_models.administrator import AdministratorRegistrationRequest
 from src.core.db.models import Administrator
 from src.core.db.repository import AdministratorRepository
+from src.core.services.administrator_invitation import AdministratorInvitationService
+from src.core.services.authentication_service import AuthenticationService
 
 
 class AdministratorService:
-    def __init__(self, administrator_repository: AdministratorRepository = Depends()):
+    def __init__(
+        self,
+        administrator_repository: AdministratorRepository = Depends(),
+        administrator_invitation_service: AdministratorInvitationService = Depends(),
+    ):
         self.__administrator_repository = administrator_repository
+        self.__administrator_invitation_service = administrator_invitation_service
+
+    async def register_new_administrator(self, token: UUID, schema: AdministratorRegistrationRequest) -> Administrator:
+        """Регистрация нового администратора."""
+        invitation = await self.__administrator_invitation_service.get_invitation_by_token(token)
+        administrator = await schema.parse_to_db_obj()
+        administrator.email = invitation.email
+        administrator.hashed_password = AuthenticationService.get_hashed_password(schema.password.get_secret_value())
+        administrator.status = Administrator.Status.ACTIVE
+        administrator.role = Administrator.Role.PSYCHOLOGIST

```

Comment:
Предлагаю собрать объект со всеми аргументами здесь. Всё в одном месте. Так проще и очевиднее.
```suggestion
        administrator = Administrator(
            name = schema.name,
            surname= schema.surname,
            email = invitation.email,
            hashed_password = AuthenticationService.get_hashed_password(schema.password.get_secret_value()),
            status = Administrator.Status.ACTIVE,
            role = Administrator.Role.PSYCHOLOGIST,
        )
```